numpy/core/src/multiarray/descriptor.c

```diff
@@ -1425,6 +1436,12 @@ _convert_from_type(PyObject *obj) {
         }
         Py_DECREF(ret);
 
+        if (DEPRECATE("Converting a type/class not known to NumPy to a dtype "
+                      "currently always returns `np.dtype(object)`. This loses "
+                      "the type information and is deprecated.") < 0) {
```

Comment:
I'm inclined not to deprecate this - it's a way to optimistically write code in a way that assumes maybe numpy will natively support the type in the distant future, and do "the right thing".

Reply:
Yeah, I was not quite sure about this one, the optimistically writing does make a lot of sense (it is also a bit of a trap), but...
If we never deprecate this, it can eventually lead to subtle behaviour changes in downstream code. Usually, to the *intended* behaviour hopefully.

I guess I will remove this part, and see if I can add a note to the dtype documentation.

I somewhat intend on deprecating `np.dtype("S")` (still allowing it for most `dtype="S"` arguments). At that point, I will probably revisit this part.